src/net/discovery/mod.rs

```diff
@@ -0,0 +1,24 @@
+pub mod config;
+#[cfg(test)]
+pub mod mock;
+
+use crate::net::client::MpcHttpConnection;
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error(transparent)]
+    ParseError(#[from] serde_json::Error),
+    #[error(transparent)]
+    IOError(#[from] std::io::Error),
+}
+
+/// Provides a set of peer helpers for an MPC computation. Also includes the client pointing to the
+/// running server. Since the running server is aware of which [`Identity`] it is (`H1`, `H2`, or
```

Comment:
should `Identity` assignment be done as part of peer discovery? or will that process happen elsewhere?

Reply:
Provided my suggestion above

Reply:
gotcha. so since we are passing `--role` as a runtime flag, we can safely exclude that information here